client/blocks/edit-gravatar/index.jsx

```diff
@@ -157,6 +159,33 @@ export class EditGravatar extends Component {
 		}
 	}
 
+	renderEmailVerificationNotice() {
+		const user = {
+			data: this.props.user
+		};
+		return (
+			<VerifyEmailDialog
+				user={ user }
+				onClose={ this.closeVerifyEmailDialog }
+			/>
+		);
+	}
+
+	handleClick = () => {
+		if ( this.props.user.email_verified ) {
+			return;
+		}
+		this.setState( () => ( {
+			showEmailVerificationNotice: true
+		} ) );
```

Comment:
although `setState()` can accept a function to return the updated keys if we're not using any values from `state` or `props` (which is normally the case) we can just pass in the object…

```js
this.setState( { showEmailVerificationNotice: true } );
```